### ISSUE TITLE
TypeError: Required argument 'flags' (pos 6) not found

### ISSUE THREAD

**Issue by TDeVries** [hdidx/hdidx#7]
Thanks for creating this package! It works great.

I'm posting this for anyone else who may run into the same issue.

I got the following error, which stems from OpenCV's KMeans function:
```
/hdidx/util.py in kmeans(vs, ks, niter)
     61         flags = cv2.KMEANS_RANDOM_CENTERS
     62         compactness, labels, centers = cv2.kmeans(
---> 63             vs, ks, criteria, 1, flags)
     64         return centers
     65 except ImportError:

TypeError: Required argument 'flags' (pos 6) not found
```
I think that in newer versions of OpenCV the KMeans function requires an extra input. My fix was to change line 62 and 63 to:

`compactness, labels, centers = cv2.kmeans(vs, ks, None, criteria, 1, flags)`

**Comment by willard-yuan** [comment]
@TDeVries Your OpenCV's version is 3.x?

**Comment by TDeVries** [comment]
Yes, I should have mentioned that. Version 3.1.0 to be specific.

**Comment by piperod** [comment]
From the[ documentation](https://docs.opencv.org/3.0-beta/doc/py_tutorials/py_ml/py_kmeans/py_kmeans_opencv/py_kmeans_opencv.html) is not clear why the 6th position is added. But looking at the examples, you can notice you only have to add one position with None. 
ret,label,center=cv2.kmeans(Z,2,**None**,criteria,10,cv2.KMEANS_RANDOM_CENTERS)

**Comment by offbye** [comment]
I use opencv 3.4,  also need fix it.
